### ISSUE TITLE
Broken Release Action?

### ISSUE THREAD

**Issue by hoeck** [moltar/typescript-runtime-type-benchmarks#1479]
After merging a PR I noticed that the [release](https://github.com/moltar/typescript-runtime-type-benchmarks/actions/runs/12347917382/job/34466895195) action fails when trying to commit the benchmark results (after having succeeded in committing the results of another node version of the matrix build).
So the results for a new or updated library are only shown for those node versions where the commit was successful. At least for a few hours until renovate-bot triggers a new release.

Running the benchmarks locally, everything runs fine. A single JSON file is written and it should not cause problems when committing that.

Then I noticed that other runs of the release action also (randomly?) [failed](https://github.com/moltar/typescript-runtime-type-benchmarks/actions/workflows/release.yml?query=is%3Afailure).

Without digging deeper into the reason I'd suggest replacing the aged https://github.com/github-actions-x commit action with something that looks a bit more maintained, e.g. https://github.com/EndBug/add-and-commit

To me this error smells like some race condition that maybe happens when there are too many commits going on in parallel?

Any ideas, suggestions or objections?

If not I would like to try changing the release workflow to use said commit action and see if it helps.

**Comment by moltar** [comment]
Yeah, there _was_ a race condition a while ago, and I thought I had fixed it but perhaps not.

No objections on replacing legacy actions with something more maintained.

**Comment by DarkGL** [comment]
Been using https://github.com/github-actions-x in my private projects, and it just stopped working.

Switched to https://github.com/EndBug/add-and-commit and everything works fine, @hoeck do you want to prepare pr, or I can do it?

**Comment by hoeck** [comment]
> Switched to https://github.com/EndBug/add-and-commit and everything works fine, @hoeck do you want to prepare pr, or I can do it?

Yes, please create a PR.

I'm unfortunately bound to a project at work that is due to the end of the year (as always).

Thanks!
